### PR TITLE
feat: [SRM-13373]: Remove Cloud watch, ELK and Hosted prometheus feature flags

### PR DESCRIPTION
**Pull request by praneshg239**
### Summary
Removed feature flags:

1. SRM_ENABLE_HEALTHSOURCE_CLOUDWATCH_METRICS
2. ELK_HEALTH_SOURCE
3. SRM_ENABLE_HEALTHSOURCE_AWS_PROMETHEUS

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by deepesh-ui — DISMISSED** [review]

File: src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts [hunk 1]
```diff
@@ -309,15 +298,11 @@ const getHealthSourceType = (type?: string, sourceType?: string): string | undef
...
-  type,
-  dataSourceType,
-  isDataSourceTypeSelectorEnabled
-}: GetDataSourceTypeParams): string | null => {
+export const getDataSourceType = ({ type, dataSourceType }: GetDataSourceTypeParams): string | null => {
   if (type !== HealthSourceTypes.NextGenHealthSource) {
     if (type === HealthSourceTypes.AwsPrometheus || dataSourceType === AWSDataSourceType) {
       return AWSDataSourceType
-    } else if (isDataSourceTypeSelectorEnabled) {
+    } else if (type === HealthSourceTypes.Prometheus || dataSourceType === HealthSourceTypes.Prometheus) {
```
Comment (harshilbits) [inline src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts:305]: we should we avoiding any condition specific to `Prometheus`

File: src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts [hunk 2]
```diff
@@ -423,14 +403,10 @@ export function getWorkspaceDropdownOptions(workspaces?: AwsPrometheusWorkspaceD
   return workspaceOptions
 }
 
-export function canShowDataSelector(sourceType?: string, isDataSourceTypeSelectorEnabled?: boolean): boolean {
-  return Boolean(sourceType === HealthSourceTypes.Prometheus && isDataSourceTypeSelectorEnabled)
+export function canShowDataSelector(sourceType?: string): boolean {
+  return Boolean(sourceType === HealthSourceTypes.Prometheus)
```
Comment (harshilbits) [inline src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts:407]: same

File: src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts (hunk 1 above)
Comment (praneshg239) [reply src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts:305]: It is only for Prometheus
image

File: src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts (hunk 2 above)
Comment (praneshg239) [reply src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts:407]: same as above

**Review by harshilbits — APPROVED** [review]

**Review by deepesh-ui — APPROVED** [review]

**Pull request merged**